### PR TITLE
Fixing support for webpack-dev-server v4

### LINKED ISSUE #893: Update to Encore 1.0 dev-server failed
Hello,

I try updating my project with the new `Encore 1.0`, but after trying to update my config I ran on a fail build

Here is the error when I start `npm run dev-server`
```
[webpack-cli] Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
 - configuration has an unknown property 'disableHostCheck'. These properties are valid:
   object { bonjour?, client?, compress?, dev?, firewall?, headers?, historyApiFallback?, host?, hot?, http2?, https?, injectClient?, injectHot?, liveReload?, onAfterSetupMiddleware?, onBeforeSetupMiddleware?, onListening?, open?, openPage?, overlay?, port?, proxy?, public?, setupExitSignals?, static?, stdin?, transportMode?, useLocalIp? }
```

Here is the npm log:

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'dev-server' ]
2 info using npm@6.14.8
3 info using node@v14.13.1
4 verbose run-script [ 'predev-server', 'dev-server', 'postdev-server' ]
5 info lifecycle @~predev-server: @
6 info lifecycle @~dev-server: @
7 verbose lifecycle @~dev-server: unsafe-perm in lifecycle true
8 verbose lifecycle @~dev-server: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/klesk/<project>/node_modules/.bin:/home/klesk/.npm-global/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
9 verbose lifecycle @~dev-server: CWD: /home/klesk/<project>
10 silly lifecycle @~dev-server: Args: [ '-c', 'encore dev-server' ]
11 silly lifecycle @~dev-server: Returned: code: 2  signal: null
12 info lifecycle @~dev-server: Failed to exec dev-server script
13 verbose stack Error: @ dev-server: `encore dev-server`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1047:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
14 verbose pkgid @
15 verbose cwd /home/klesk/<project>
16 verbose Linux 4.19.128-microsoft-standard
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "dev-server"
18 verbose node v14.13.1
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 2
22 error @ dev-server: `encore dev-server`
22 error Exit status 2
23 error Failed at the @ dev-server script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

I try to remove `node_module` and npm install with the same error.

But `npm run build` works with the same configuration.

Thanks

### PR DESCRIPTION
**Pull request by weaverryan**
Fixes #893

### CODE REVIEW

File: lib/config/parse-runtime.js [hunk 1]
```diff
@@ -42,7 +42,6 @@ module.exports = function(argv, cwd) {
 
             runtimeConfig.useDevServer = true;
             runtimeConfig.devServerHttps = argv.https;
-            runtimeConfig.useHotModuleReplacement = argv.hot || false;
```
Comment (weaverryan) [inline lib/config/parse-runtime.js:45]: `hot` now defaults to true. So there is no option at the command line and no need for us to set anything.

File: lib/config-generator.js [hunk 2]
```diff
@@ -570,17 +570,16 @@ class ConfigGenerator {
...
+                publicPath: this.webpackConfig.getRealPublicPath(),
+            },
             // avoid CORS concerns trying to load things like fonts from the dev server
             headers: { 'Access-Control-Allow-Origin': '*' },
-            hot: this.webpackConfig.useHotModuleReplacementPlugin(),
             // required by FriendlyErrorsWebpackPlugin
-            quiet: true,
             compress: true,
             historyApiFallback: true,
-            watchOptions: this.buildWatchOptionsConfig(),
```
Comment (weaverryan) [inline lib/config-generator.js:583]: This now lives under `static.watch`. However, it automatically reads the root-level `watchOptions` config and uses that as its defaults - https://github.com/webpack/webpack-dev-server/blob/788f2ba7d9152610ba0d3024f0079c98aad86991/lib/utils/normalizeOptions.js#L9-L23 - so we don't need to re-set the options here also.

File: lib/config-generator.js [hunk 3]
```diff
@@ -570,17 +570,16 @@ class ConfigGenerator {
...
-            publicPath: this.webpackConfig.getRealPublicPath(),
+            static: {
+                directory: contentBase,
+                // this doesn't appear to be necessary, but here in case
+                publicPath: this.webpackConfig.getRealPublicPath(),
+            },
             // avoid CORS concerns trying to load things like fonts from the dev server
             headers: { 'Access-Control-Allow-Origin': '*' },
-            hot: this.webpackConfig.useHotModuleReplacementPlugin(),
             // required by FriendlyErrorsWebpackPlugin
```
Comment (stof) [inline lib/config-generator.js:580]: this comment should be removed, as you removed the option that was explained by the comment.

Comment (weaverryan) [reply lib/config-generator.js]: thanks for catching that!

**Pull request merged**